sync.js

```diff
@@ -463,13 +464,21 @@ function WifiPeer (connection, info) {
 // key is String or Buffer
 function hash (key) {
   if (typeof key === 'string') {
-    key = Buffer.from(key, 'hex')
-  }
-  if (Buffer.isBuffer(key) && key.length === 32) {
-    return crypto.discoveryKey(key).toString('hex')
-  } else {
-    return null
+    key = Buffer.from(key)
   }
+  if (!Buffer.isBuffer(key) || key.length === 0) return null
+  return discoveryKey(key)
+}
+
+// 32-byte key used for generating the discovery key
+const MAPEO = bufferAlloc(32)
+sodium.crypto_generichash(MAPEO, bufferFrom('mapeo'))
```

Comment:
`Buffer.from` and `Buffer.allocUnsafe` were added in node 5.10.0, so they should be safe to use /wo polyfills.